### PR TITLE
feat: calculate usd price from pools via thornode

### PR DESCRIPTION
**Pull request by kaladinlight**
- midgard is slow replace with thornode
- thornode does not have usd price immediately available, calculate from pool ratios
- get average rune usd price from all available usd pools for most accurate price representation
- set stage for improved function signatures via class variables (eventually want to pull in helper functions to reduce file bloat)
- small cleanup/performance improvement for tracking available pools on `swapper.initialize()`

### CODE REVIEW

File: packages/swapper/src/swappers/thorchain/ThorchainSwapper.ts [hunk 1]
```diff
@@ -65,12 +68,21 @@ export class ThorchainSwapper implements Swapper<ChainId> {
...
+  midgardUrl: string
+  adapterManager: ChainAdapterManager
+  web3: Web3
 
   constructor(deps: ThorchainSwapperDeps) {
     this.deps = deps
+    this.daemonUrl = deps.daemonUrl
+    this.midgardUrl = deps.midgardUrl
+    this.adapterManager = deps.adapterManager
+    this.web3 = deps.web3
```
Comment (gomesalexandre) [inline packages/swapper/src/swappers/thorchain/ThorchainSwapper.ts:85]: Do we need to inject this? Doesn't seem used anywhere

**Review by gomesalexandre — COMMENTED** [review]
Conceptually looks good to me, though comment-as-a-stamp (CaaS), untested as I'm in a different domain currently

File: packages/swapper/src/swappers/thorchain/ThorchainSwapper.ts (hunk 1 above)
Comment (kaladinlight) [reply packages/swapper/src/swappers/thorchain/ThorchainSwapper.ts:85]: not yet, just setting the stage to use class variables and pull in helper functions to improve readability and maintainability of these swappers

Comment (kaladinlight) [reply packages/swapper/src/swappers/thorchain/ThorchainSwapper.ts:85]: i can remove until used I suppose, just want to start cleaning these up some, but didn't want to add too much to this pr.

**Review by 0xApotheosis — APPROVED** [review]
Tested whilst linked locally, LGTM - nice tidy up and optimisation.

**Pull request merged**